src/Core/Services/Implementations/CipherService.cs

```diff
@@ -288,13 +288,24 @@ public async Task DeleteAsync(Cipher cipher, Guid deletingUserId, bool orgAdmin
             await _pushService.PushSyncCipherDeleteAsync(cipher);
         }
 
-        public async Task DeleteManyAsync(IEnumerable<Guid> cipherIds, Guid deletingUserId)
+        public async Task DeleteManyAsync(IEnumerable<Guid> cipherIds, Guid deletingUserId, Guid? organizationId = null, bool orgAdmin = false)
         {
+
             var cipherIdsSet = new HashSet<Guid>(cipherIds);
-            var ciphers = await _cipherRepository.GetManyByUserIdAsync(deletingUserId);
-            var deletingCiphers = ciphers.Where(c => cipherIdsSet.Contains(c.Id) && c.Edit);
+            var deletingCiphers = new List<Cipher>();
+
+            if (orgAdmin && organizationId.HasValue)
+            {
+                var ciphers = await _cipherRepository.GetManyByOrganizationIdAsync(organizationId.Value);
+                deletingCiphers = ciphers.Where(c => cipherIdsSet.Contains(c.Id)).ToList();
+            }
+            else
+            {
+                var ciphers = await _cipherRepository.GetManyByUserIdAsync(deletingUserId);
+                deletingCiphers = ciphers.Where(c => cipherIdsSet.Contains(c.Id) && c.Edit).Select(x => (Cipher)x).ToList();
+            }
 
-            await _cipherRepository.DeleteAsync(cipherIds, deletingUserId);
+            await _cipherRepository.DeleteAsync(deletingCiphers.Select(x => x.Id), deletingUserId);
```

Comment:
I still think there is a gap here. This sproc doesn't appear to handle admin cases. Same with the soft delete.

Reply:
Implemented based on the solution we talked about off camera.